src/pages/ExamsPage/data/reducer.js

```diff
@@ -70,6 +70,20 @@ const slice = createSlice({
         return data;
       }),
     }),
+    editAllowance: (state, { payload }) => ({
```

Comment:
Added the reducer for editAllowance so we're not fetching all allowances every time (which is what I was doing before).